### PR TITLE
Run build-artifacts from Ubuntu and Windows in the emulator

### PR DESCRIPTION
**Pull request by MarijnS95**
For https://github.com/rust-windowing/android-ndk-rs/pull/162#discussion_r683810422

~TODO: We're not testing `cargo apk run` functionality anymore, how do we want to work that into here? We can place the commands inside `rust.yml` in a third emulator run.~ This is now happening in a third `local` run.

Note that this PR is intentionally _not_ based on the merge of #168 to demonstrate it actually failing! This would have caught the bogus refactor in #162.

### CODE REVIEW

File: cargo-apk/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "cargo-apk"
-version = "0.8.1"
+version = "0.8.0"
```
Comment (dvc94ch) [inline cargo-apk/Cargo.toml:3]: this seems like a bogus change

File: ndk-build/src/apk.rs [hunk 2]
```diff
@@ -84,15 +84,8 @@ impl<'a> UnalignedApk<'a> {
...
-        // Pass UNIX path separators to `aapt` on non-UNIX systems, ensuring the resulting separator
-        // is compatible with the target device instead of the host platform.
-        // Otherwise, it results in a runtime error when loading the NativeActivity `.so` library.
-        let lib_path_unix = lib_path.to_str().unwrap().replace("\\", "/");
-
         let mut aapt = self.0.build_tool(bin!("aapt"))?;
-        aapt.arg("add")
-            .arg(self.0.unaligned_apk())
-            .arg(lib_path_unix);
+        aapt.arg("add").arg(self.0.unaligned_apk()).arg(&lib_path);
```
Comment (dvc94ch) [inline ndk-build/src/apk.rs:88]: I guess there is a reason for this and was thoroughly tested

**Review by dvc94ch — APPROVED** [review]
LGTM, thanks!

**Comment by MarijnS95** [comment]
@dvc94ch immediately resolving your comments: I _intentionally_ pushed a [revert of](https://github.com/rust-windowing/android-ndk-rs/pull/169/commits/577305dea9e271d2df46c0cacf20318b86fd94db) @msiglreith's [fix](https://github.com/rust-windowing/android-ndk-rs/pull/168) to showcase that this improved Emulator setup [indeed catches that issue](https://github.com/rust-windowing/android-ndk-rs/runs/3274663232?check_suite_focus=true) :)

**Comment by dvc94ch** [comment]
looks good now

**Comment by MarijnS95** [comment]
Now that we have a failure case clearly visible and linked: https://github.com/rust-windowing/android-ndk-rs/runs/3274663232?check_suite_focus=true this PR is now running with the fix in place and should succeed. Ready for review and merge :)

**Review by msiglreith — APPROVED** [review]

**Comment by MarijnS95** [comment]
The script strangely has a tendency to get stuck, on a different `matrix.source_os` even. Added `set -x` to figure out what's going on. It gets stuck here:

```log
2021-08-08T18:21:16.0576230Z Emulator booted.
2021-08-08T18:21:16.0599730Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell input keyevent 82
2021-08-08T18:21:16.9240320Z Disabling animations.
2021-08-08T18:21:16.9277890Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell settings put global window_animation_scale 0.0
2021-08-08T18:21:17.4115870Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell settings put global transition_animation_scale 0.0
2021-08-08T18:21:17.6868170Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell settings put global animator_duration_scale 0.0
2021-08-08T18:21:17.7762760Z [command]/bin/sh -c ./.github/workflows/android_test.sh "ubuntu-latest" "/Users/runner/work/android-ndk-rs/android-ndk-rs"
2021-08-08T18:21:17.9564350Z emulator: INFO: boot completed
2021-08-08T18:21:17.9567220Z emulator: INFO: boot time 64960 ms
2021-08-08T18:21:17.9567880Z emulator: Increasing screen off timeout, logcat buffer size to 2M.
2021-08-08T18:21:17.9580040Z emulator: Revoking microphone permissions for Google App.
2021-08-08T18:21:43.0403370Z emulator: ### WARNING: /etc/localtime does not point to zoneinfo-compatible timezone name
2021-08-08T18:21:43.0505090Z 
2021-08-08T20:39:47.6888850Z ##[error]The operation was canceled.
```
A working build looks like:
```log
2021-08-08T18:21:08.9113400Z Emulator booted.
2021-08-08T18:21:08.9168200Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell input keyevent 82
2021-08-08T18:21:11.1096720Z emulator: INFO: boot completed
2021-08-08T18:21:11.1198330Z emulator: INFO: boot time 53960 ms
2021-08-08T18:21:11.1299610Z emulator: Increasing screen off timeout, logcat buffer size to 2M.
2021-08-08T18:21:11.1396130Z emulator: Revoking microphone permissions for Google App.
2021-08-08T18:21:11.9884620Z Disabling animations.
2021-08-08T18:21:11.9907410Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell settings put global window_animation_scale 0.0
2021-08-08T18:21:12.3787490Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell settings put global transition_animation_scale 0.0
2021-08-08T18:21:12.5684550Z [command]/Users/runner/Library/Android/sdk/platform-tools/adb shell settings put global animator_duration_scale 0.0
2021-08-08T18:21:12.7914430Z [command]/bin/sh -c ./.github/workflows/android_test.sh "local" ""
2021-08-08T18:21:13.3565060Z Warning: You use environment variable ANDROID_HOME that is deprecated. Please, remove it and use ANDROID_SDK_ROOT instead. Now ANDROID_HOME is used
2021-08-08T18:21:14.5924190Z     Updating crates.io index
... andsoforth
```

**Comment by MarijnS95** [comment]
```
Warning: An error occurred while preparing SDK package Intel x86 Atom_64 System Image: Operation timed out (Read failed).
/Users/runner/Library/Android/sdk/platform-tools/adb -s emulator-5554 emu kill
error: could not connect to TCP port 5554: Connection refused
The process '/Users/runner/Library/Android/sdk/platform-tools/adb' failed with exit code 1
```
Ehh, let's try one more time...?

Is this something preventing us from running >2 emulators in a single CI run? None of these errors appear related to our scripts at all.

**Pull request merged**